_includes/_code/lakes-interactive/lib/border-star-button.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/material.dart';
+// Uncomment lines 3 and 6 to view the visual layout at runtime.
+//import 'package:flutter/rendering.dart' show debugPaintSizeEnabled;
+
+void main() {
+  //debugPaintSizeEnabled = true;
+  runApp(new MyApp());
+}
+
+class FavoriteWidget extends StatefulWidget {
+  @override
+  FavoriteWidgetState createState() => new FavoriteWidgetState();
+}
+
+class FavoriteWidgetState extends State<FavoriteWidget> {
```

Comment:
Yeah, I'm not suggesting with use private in all the samples.  This tutorial is specifically about managing state, which is about how to make a `State` and how to deal with `setState`.  The best practice for making a `State` object is to make the class private and the best practice for member variables of a `State` object is for them to be private and mutated only within a `setState` callback.  If we don't teach this information in this tutorial, I'm not sure where else we'd teach it.